src/api.rs

```diff
@@ -33,16 +33,110 @@ pub struct EncoderConfig {
   pub key_frame_interval: u64,
   pub low_latency: bool,
   pub quantizer: usize,
-  pub speed: usize,
-  pub tune: Tune
+  pub tune: Tune,
+  pub min_block_size: BlockSize,
+  pub multiref: bool,
+  pub fast_deblock: bool,
+  pub reduced_tx_set: bool,
+  pub tx_domain_distortion: bool,
+  pub encode_bottomup: bool,
+  pub rdo_tx_decision: bool,
+  pub prediction_modes: PredictionModesSetting,
+  pub include_near_mvs: bool,
 }
 
 impl Default for EncoderConfig {
   fn default() -> Self {
-    EncoderConfig { key_frame_interval: 30, low_latency: true, quantizer: 100, speed: 0, tune: Tune::Psnr  }
+    const DEFAULT_SPEED: usize = 3;
+    Self::with_speed_preset(DEFAULT_SPEED)
```

Comment:
This would loop infinitely

Reply:
You may have a `SpeedSettings` struct with the standard default impl and embed it in the `EncoderConfig`.

Reply:
AWESOME IDEA!